main.cpp

```diff
@@ -328,11 +244,16 @@ struct IntType
     IntType &subtract(int rhs);
     IntType &multiply(int rhs);
     IntType &divide(int rhs);
+    IntType &pow(int exp);
+    IntType &pow(const FloatType &exp);
+    IntType &pow(const DoubleType &exp);
+    IntType &pow(const IntType &exp);
```

Comment:
Don't put `&` next to the function name. 
put it next to the return type.
It looks like you're taking the address of the function, instead of returning a reference.
- [ ] revise